File: vbnet/simple_calculator/simple_calculator_11.vb
' This program is a simple calculator written in Visual Basic .NET.
' It performs basic arithmetic operations such as addition, subtraction,
' multiplication, and division.

Module SimpleCalculator

    Sub Main()
        Dim num1 As Double
        Dim num2 As Double
        Dim result As Double
        Dim operation As String

        Console.WriteLine("Enter the first number:")
        num1 = Convert.ToDouble(Console.ReadLine())

        Console.WriteLine("Enter the second number:")
        num2 = Convert.ToDouble(Console.ReadLine())

        Console.WriteLine("Enter the operation (+, -, *, /):")
        operation = Console.ReadLine()

        Select Case operation
            Case "+"
                result = num1 + num2
            Case "-"
                result = num1 - num2
            Case "*"
                result = num1 * num2
            Case "/"
                result = num1 / num2
            Case Else
                Console.WriteLine("Invalid operation")
                Exit Sub
        End Select

        Console.WriteLine("The result is: " & result)
    End Sub

End Module

File: swift/simple_calculator/simple_calculator_11.swift
// This program is a simple calculator that performs basic arithmetic operations.
// It is designed to help programmers practice their Swift programming skills.

import Foundation

func add(_ a: Double, _ b: Double) -> Double {
    return a + b
}

func subtract(_ a: Double, _ b: Double) -> Double {
    return a - b
}

func multiply(_ a: Double, _ b: Double) -> Double {
    return a * b
}

func divide(_ a: Double, _ b: Double) -> Double {
    return a / b
}

print("Enter first number:")
guard let firstInput = readLine(), let firstNumber = Double(firstInput) else {
    print("Invalid input")
    exit(1)
}

print("Enter second number:")
guard let secondInput = readLine(), let secondNumber = Double(secondInput) else {
    print("Invalid input")
    exit(1)
}

print("Enter operation (+, -, *, /):")
guard let operation = readLine() else {
    print("Invalid input")
    exit(1)
}

var result: Double?

switch operation {
case "+":
    result = add(firstNumber, secondNumber)
case "-":
    result = subtract(firstNumber, secondNumber)
case "*":
    result = multiply(firstNumber, secondNumber)
case "/":
    result = divide(firstNumber, secondNumber)
default:
    print("Invalid operation")
    exit(1)
}

if let result = result {
    print("Result: \(result)")
} else {
    print("Error calculating result")
}

